### DISCUSSION TITLE
[@types/react-simple-oauth2-login] children prop potentially declared with wrong type.

### DISCUSSION THREAD

**Discussion by Krelsis** [DefinitelyTyped/DefinitelyTyped#65251] [Issues with a @types package]
Currently when trying to nest children elements as seen in the snippet below, the following error is thrown `TS2739	(TS) Type 'ReactElement<any, any>' is missing the following properties from type 'ReactChildren': map, forEach, count, only, toArray`

```jsx
<OAuth2Login
  className="OAuth2LoginWithChildren"
  authorizationUrl={AuthSettings["URL"] || ""}
  clientId={AuthSettings["ClientId"] || ""}
  redirectUri={AuthSettings["RedirectURL"] || ""}
  responseType="token"
  scope={AuthSettings["Scope"] || ""}
  onSuccess={OnSuccess}
  onFailure={OnFailure}
  popupHeight={860}
  popupWidth={540}
>
  <div
    id="LoginProviderSeviceA"
    className="LoginProviderServiceIcon"
    onClick={() => { BeginLogin(ServiceProviderEnum.PrimaryService); }}
    alt="Login with the primary login provider"
  />
</OAuth2Login>
```
OR
```jsx
<OAuth2Login
  {...args}
>
  <>
    <div
      id="LoginProviderSeviceA"
      className="LoginProviderServiceIcon"
      onClick={() => { BeginLogin(ServiceProviderEnum.ServiceA); }}
      alt="Login with service A login provider"
    />
    <div
      id="LoginProviderSeviceB"
      className="LoginProviderServiceIcon"
      onClick={() => { BeginLogin(ServiceProviderEnum.ServiceB); }}
      alt="Login with service B login provider"
    />
  </>
</OAuth2Login>
```
The current [definition](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-simple-oauth2-login/index.d.ts) of `children` is [React.ReactChildren](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/be21e88d0d7bd8d3714ae3af1baf9f9f925e456e/types/react-simple-oauth2-login/index.d.ts#L17).

Changing the definition of `children?: React.ReactChildren;` to `children?: ReactNode;` will fix this error. Additionally, I have even gone ahead and changed this type definition locally and all works fine, thought it may be worth suggesting this made be made a permanent change.

Finally, having glanced at the [source code](https://github.com/bhubr/react-simple-oauth2-login/blob/master/src/OAuth2Login.jsx) the component makes no effort to use any methods implemented by react.ReactChildren, additionally, further towards the [bottom of the file, in the proptypes property](https://github.com/bhubr/react-simple-oauth2-login/blob/f3c989c6a21c85759d566e6f2eabb8164aecdde1/src/OAuth2Login.jsx#L144); children is declared as a node; as such, this change is unlikely to be a breaking change.

It would probably be a good idea to implement this change for ease of use as well as the fact that many other packages in react, as well as the [react definitions](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/be21e88d0d7bd8d3714ae3af1baf9f9f925e456e/types/react/index.d.ts#L1386) itself declare `ReactNode` as the type for `children?`.

This would need to be updated within the [render method](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/be21e88d0d7bd8d3714ae3af1baf9f9f925e456e/types/react-simple-oauth2-login/index.d.ts#L24) also.

[a comment by an automated account is not shown]

**Comment by MysteryBlokHed** [comment]
I just opened #65273
